src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -82,11 +91,53 @@ object WdlExpression {
     val parseTree = parser.parse_e(tokens, new WdlSyntaxErrorFormatter(terminalMap))
     new WdlExpression(parseTree.toAst)
   }
+
+  def toString(ast: AstNode, highlighter: SyntaxHighlighter = NullSyntaxHighlighter): String = {
```

Comment:
On your last point note that he's a FP purist speaking to a very FP-oriented crowd where cracks about common imperative and OO patterns are legion. 

Also I don't think either of us were suggesting that you specifically adopt the Reader approach. Comments on pull requests have a way of not necessarily being direct commentary on the code in the pull request.